3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,218 @@
+    return;
+  }
+
+  
+  for (const note of notes) {
+    if (note.id === id) {
+      return note;
+    }
+  }
+
+  
+  console.log("Error: Note not found.");
+}
+
+function logOutNotesFormatted() {
+  for (const note of notes) {
+    console.log(`The note with id: ${note.id}, has the following note text: ${note.content}`);
+  }
+}
+
+/
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+logOutNotesFormatted();
+// should log out:
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+
+
+  //(4) - CactusIO-interactive (Smart phone usage app) optional// 
+
+  
+const activities = [];
+let usageLimit = 0;
+
+// Function to add an activity.
+function addActivity(activity, duration) {
+  // Get the current date in the 'dd/mm-yy' format.
+  const currentDate = new Date().toLocaleDateString('en-GB').replace(/\//g, '-');
+  
+  // Push the activity object to the activities array.
+  activities.push({
+    date: currentDate,
+    activity,
+    duration,
+  });
+}
+
+// Function to show the status.
+function showStatus() {
+  if (activities.length === 0) {
+    console.log("Add some activities before calling showStatus.");
+    return;
+  }
+
+  const today = new Date().toLocaleDateString('en-GB').replace(/\//g, '-');
+  const todayActivities = activities.filter(activity => activity.date === today);
+  const totalDuration = todayActivities.reduce((sum, activity) => sum + activity.duration, 0);
+  
+  console.log(`You have added ${todayActivities.length} activities. They amount to ${totalDuration} min. of usage`);
+  
+ 
+  if (usageLimit > 0 && totalDuration > usageLimit) {
+    console.log("You have reached your limit, no more smartphoning for you!");
+  }
+}
+
+function setUsageLimit(limit) {
+  usageLimit = limit;
+}
+
+function mostTimeSpentActivity() {
+  if (activities.length === 0) {
+    console.log("No activities added yet.");
+    return;
+  }
+  
+  const activityMap = new Map();
+  
+  activities.forEach(activity => {
+    if (activityMap.has(activity.activity)) {
+      activityMap.set(activity.activity, activityMap.get(activity.activity) + activity.duration);
+    } else {
+      activityMap.set(activity.activity, activity.duration);
+    }
+  });
+  
+  const mostTimeSpent = [...activityMap.entries()].reduce((a, b) => (a[1] > b[1] ? a : b));
+  
+  console.log(`You have spent the most time on ${mostTimeSpent[0]} with ${mostTimeSpent[1]} min in day.`);
+}
+
+addActivity("Youtube", 30);
+addActivity("Facebook", 15);
+addActivity("News", 10);
+addActivity("Instagram", 25);
+
+setUsageLimit(60); // Set the usage limit to 60 minutes
+
+showStatus(); // Show the status
+mostTimeSpentActivity(); // Find the activity with the most time spent
+
```

Comment:
🌟 Your implementation for getting the current date in the desired format is spot on! Also, the dynamic way of fetching activities for the current date and summarizing them in showStatus is well executed.
💡 Suggestion: In `mostTimeSpentActivity`, since the activities are dynamic, maybe add a comment to notify users that the outcome might differ based on the day's activities.